Framework/API/inc/MantidAPI/IFunction.h

```diff
@@ -301,6 +364,33 @@ class MANTID_API_DLL IFunction {
     mutable boost::variant<std::string, int, double, bool, std::vector<double>> m_data;
     /// Flag indicating if the string value should be returned quoted
     bool m_quoteValue = false;
+    /// Associated Validator
+    mutable Kernel::IValidator_sptr m_validator;
+  };
+
+  /**
+   * Atribute validator visitor class. It is used to access
+   * m_data of a specificed attribute in order to
+   * evaluate the current value of the attribute against
+   * the associated validator, for all allowed types.
+   */
+  template <typename T = void> class DLLExport AttributeValidatorVisitor : public boost::static_visitor<T> {
```

Comment:
I'm not sure it makes much sense to do this, as the class is only applicable to function attributes which are nested within the IFunction class. It makes sense to me that this class is therefore also within the IFunction class?

Reply:
We could put something similar in the doc string? Unless this will make it difficult
I'm just concerned we keep including more and more into IFunction over time. Equally I'm happy to merge this as-is now too

Reply:
I agree IFunction is getting unwieldly, largely due to the nested classes of which I personally not a fan! In the future it's probably worth considering separating out Function Attributes and related structs/classes into their own header. There's another few (smaller) PRs to follow this one, so I think merge as-is and I take this forwards as a separate issue to consider?

Reply:
Agreed, I'll retest and merge if everything still works
Thanks for going through all of this